### DISCUSSION TITLE
Get full url using Inertia

### DISCUSSION THREAD

**Discussion by vz-spr** [inertiajs/inertia#1292] [Help (Vue)]
Is there possible to get full url, including protocol, host, uri, search params

Need for SSR

The only way I found so far is get it as prop from backend (in my case it's Laravel).

**Comment by Douglasokolaa** [comment]
You can simply do a `window.location` https://developer.mozilla.org/en-US/docs/Web/API/Window/location

**Reply by vz-spr** [reply]
@Douglasokolaa not in my case.

I need to use it in Server Side Rendering

In nodejs there is no `window.location`

**Reply by 300code** [reply]
import {  router, usePage  } from '@inertiajs/vue3' 
 const page = usePage();
 watchEffect(() => { 
      console.log('page', page.url)


:}

**Comment by rvanasperen** [comment]
I ran into the same issue and did some debugging.

There's a possibility of adding `Inertia::resolveUrlUsing(fn (Request $request) => $request->fullUrl());` to a service provider, which works for the initial Inertia request to return an absolute URL instead of a relative URL for `usePage().url`.

However, subsequent Inertia-powered requests (using the X-Inertia header) end up calling a `pageUrl` JS function in the Repsonse class, which strips away the hostname, rendering the aforementioned PHP call (for at least absolute URLs specifically) moot.

https://github.com/inertiajs/inertia/blob/32266646c9b64a40dd0dc78ef78833f96e7b8731/packages/core/src/response.ts#L252

The best bet is probably using a separate prop from the backend, unfortunately (i.e. add `'url' => $request->fullUrl(),` in `HandleInertiaRequests::share()`.
